### PR TITLE
Mf/validate stake lockup period

### PR DESCRIPTION
**Pull request by mattfoley8**

### CODE REVIEW

File: lib/block_view_stake.go [hunk 1]
```diff
@@ -1476,8 +1481,9 @@ func (bav *UtxoView) _connectUnstake(
 		Type:                       OperationTypeUnstake,
 		PrevValidatorEntry:         prevValidatorEntry,
 		PrevGlobalStakeAmountNanos: prevGlobalStakeAmountNanos,
-		PrevStakeEntries:           []*StakeEntry{prevStakeEntry},
-		PrevLockedStakeEntries:     []*LockedStakeEntry{prevLockedStakeEntry},
+		PrevStakeEntries:           prevStakeEntries,
+		PrevLockedStakeEntries:     prevLockedStakeEntries,
+		PrevEpochNumber:            currentEpochNumber,
```
Comment (lazynina) [inline lib/block_view_stake.go:1486]: Why do we need `PrevEpochNumber` in the utxo op? Why can't we look it up in disconnect? jw

File: lib/constants.go [hunk 2]
```diff
@@ -968,6 +972,9 @@ var DeSoMainnetParams = DeSoParams{
 	// reserve ratios.
 	CreatorCoinAutoSellThresholdNanos: uint64(10),
 
+	// Unstaked stake can be unlocked after a minimum of N elapsed epochs.
+	StakeLockupEpochDuration: uint64(3),
```
Comment (lazynina) [inline lib/constants.go:976]: will we want to move this to be a param updater thing eventually or nah?

File: lib/block_view_stake.go (hunk 1 above)
Comment (mattfoley8) [reply lib/block_view_stake.go:1486]: I think you're right. If we ever disconnect a transaction, it'll be in the same block/epoch as the connect logic was ran. I removed.

File: lib/constants.go (hunk 2 above)
Comment (mattfoley8) [reply lib/constants.go:976]: Yes, we will. I added a comment. Good call-out.

**Review by lazynina — APPROVED** [review]

**Pull request merged**